int32 height_input
